lib/libzfs/libzfs_changelist.c

```diff
@@ -541,15 +542,16 @@ change_one(zfs_handle_t *zhp, void *data)
 			uu_avl_insert(clp->cl_tree, cn, idx);
 		} else {
 			free(cn);
-			zfs_close(zhp);
 		}
 
-		if (!clp->cl_alldependents)
-			return (zfs_iter_children(zhp, change_one, data));
-	} else {
-		zfs_close(zhp);
+		if (!clp->cl_alldependents) {
+			ret = zfs_iter_children(zhp, change_one, data);
+			zfs_close(zhp);
```

Comment:
I'm pretty sure you don't want this `zfs_close()` call here.  The expectation is the `change_one` will close it in the non `zfs_iter_children` cases.

Reply:
That's possible. I haven't finished the automated internal testing for the new changes, but I'm seeing some suspicious failures.

Looking at the code though, I don't see where would we close that handle otherwise, so this is quite confusing.

Reply:
I think the `zfs_close` is necessary since `zfs_iter_children` will allocate a new `zfs_handle_t` and that is what `change_one` should close.

Reply:
So I've run `zfs-precommit` twice and the `zfs-tests` job timed out twice, at different times. There's also a bunch of failures that looked suspicious.

I've also run `zfs_precommit` twice on the code without the close after `zfs_iter_children()`. It seems passing, although both times it failed in `rsend_013_pos`, which should probably be investigated as well.

So this needs more investigation.